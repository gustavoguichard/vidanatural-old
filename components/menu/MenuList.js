import Router, { withRouter } from 'next/router'
import Link from 'next/link'
import get from 'lodash/get'
import map from 'lodash/map'
import { scrollToId } from 'utils/helpers'
import pose, { PoseGroup } from 'react-pose'

const InnerLink = pose.span({
  enter: { opacity: 1, y: 0 },
  exit: { opacity: 0, y: -15 },
})

const MenuLink = ({ href, onClick, children, ...props }) => {
  const clickHandler = event => {
    const target = get(event, 'currentTarget.href', '')
    const urlArray = target.split('#')
    const [, hash] = urlArray
    onClick(event)
    if (!hash && !target.includes('#')) {
      Router.push(target)
    }
    scrollToId(hash)
  }
  return (
    <InnerLink>
      <Link href={href}>
        <a {...props} onClick={clickHandler}>
          {children}
        </a>
      </Link>
    </InnerLink>
  )
}

const links = {
  Home: '/',
  Sobre: '/sobre',
  'Eu uso': '/eu-uso',
  Produtos: '/produtos',
  Contato: '#contato',
}

const MenuList = ({ onClick, router }) => {
  if (router.pathname === '/') {
    links.Home = '#'
    links.Sobre = '#sobre'
    links['Eu uso'] = '#eu-uso'
  }
  return (
    <>
      <nav className="menu-list-wrapper">
        <PoseGroup>
          {map(links, (path, key) => (
            <MenuLink key={key} href={path} onClick={onClick}>
              {key}
            </MenuLink>
          ))}
        </PoseGroup>
      </nav>
      <img src="/static/vine.png" alt="Ramo" width="70" />
    </>
  )
}

export default withRouter(MenuList)
