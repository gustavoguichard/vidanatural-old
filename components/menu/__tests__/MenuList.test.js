import { render } from 'react-testing-library'
import MenuList from '../MenuList'

it('renders correctly', () => {
  const { container } = render(<MenuList router={{ pathname: '/' }} />)
  expect(container).toMatchSnapshot()
})
