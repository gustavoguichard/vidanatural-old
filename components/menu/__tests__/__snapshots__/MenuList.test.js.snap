// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    class="menu-list-wrapper"
  >
    <span
      style="transform: ;transform: none;"
    >
      <a
        _pose="enter"
        href="#"
      >
        Home
      </a>
    </span>
    <span
      style="transform: ;transform: none;"
    >
      <a
        _pose="enter"
        href="#sobre"
      >
        Sobre
      </a>
    </span>
    <span
      style="transform: ;transform: none;"
    >
      <a
        _pose="enter"
        href="#eu-uso"
      >
        Eu uso
      </a>
    </span>
    <span
      style="transform: ;transform: none;"
    >
      <a
        _pose="enter"
        href="/produtos"
      >
        Produtos
      </a>
    </span>
    <span
      style="transform: ;transform: none;"
    >
      <a
        _pose="enter"
        href="#contato"
      >
        Contato
      </a>
    </span>
  </nav>
  <img
    alt="Ramo"
    src="/static/vine.png"
    width="70"
  />
</div>
`;
